File: mylib-cache-redis/src/main/java/com/zy/mylib/cache/redis/RedisUserCache.java
/*
 * Copyright © ${project.inceptionYear} ismezy (ismezy@qq.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.zy.mylib.cache.redis;

import org.springframework.data.redis.core.RedisOperations;

import java.io.Serializable;
import java.util.Map;
import java.util.concurrent.TimeUnit;

/**
 * @author ASUS
 */
public class RedisUserCache {
    private long expire;
    private TimeUnit timeUnit;
    private RedisOperations<String, ? extends Serializable> redisOperations;

    /**
     * new rediscache
     */
    public RedisUserCache(RedisOperations<String, ? extends Serializable> redisOperations, long expire, TimeUnit timeUnit) {
        this.expire = expire;
        this.timeUnit = timeUnit;
        this.redisOperations = redisOperations;
    }

    public <T extends Serializable> T get(String token, String key) {
        updateTTL(token);
        return (T) redisOperations.opsForHash().get(wrapperToken(token), key);
    }

    public <T extends Serializable> void put(String token, String key, T value) {
        updateTTL(token);
        redisOperations.opsForHash().put(wrapperToken(token), key, value);
    }

    public <T extends Serializable> void put(String token, Map<String, T> map) {
        updateTTL(token);
        redisOperations.opsForHash().putAll(wrapperToken(token), map);
    }

    public void removeToken(String token) {
        redisOperations.delete(wrapperToken(token));
    }

    private void updateTTL(String token) {
        redisOperations.expire(wrapperToken(token), expire, timeUnit);
    }

    String wrapperToken(String token) {
        return "userToken:" + token;
    }
}

File: mylib-cache-redis/src/main/kotlin/com/zy/mylib/cache/redis/RedisUserCache.kt
/**
 * Copyright © ${project.inceptionYear} ismezy (ismezy@qq.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.zy.mylib.cache.redis

import org.springframework.data.redis.core.RedisOperations
import org.springframework.cache.support.SimpleValueWrapper
import java.io.Serializable
import java.lang.RuntimeException
import java.util.concurrent.TimeUnit

/**
 * @author ASUS
 */
class RedisUserCache
/**
 * new rediscache
 */(private val redisOperations: RedisOperations<String, out Serializable>, private val expire: Long, private val timeUnit: TimeUnit) {
  operator fun <T : Serializable?> get(token: String, key: String?): T {
    updateTTL(token)
    return redisOperations.opsForHash<Any, Any>()[wrapperToken(token), key] as T
  }

  fun <T : Serializable?> put(token: String, key: String, value: T) {
    updateTTL(token)
    redisOperations.opsForHash<Any, Any>().put(wrapperToken(token), key, value)
  }

  fun <T : Serializable?> put(token: String, map: Map<String, T>?) {
    updateTTL(token)
    redisOperations.opsForHash<Any, Any>().putAll(wrapperToken(token), map)
  }

  fun removeToken(token: String) {
    redisOperations.delete(wrapperToken(token))
  }

  private fun updateTTL(token: String) {
    redisOperations.expire(wrapperToken(token), expire, timeUnit)
  }

  fun wrapperToken(token: String): String {
    return "userToken:$token"
  }
}